src/node.ts

```diff
@@ -0,0 +1,184 @@
+class ReferencedValue {
+	private dataPointer: MemoryPointer;
+	public value: Buffer;
+
+	constructor(dataPointer: MemoryPointer, value: Buffer) {
+		this.dataPointer = dataPointer;
+		this.value = value;
+	}
+}
+
+export class MemoryPointer {
+	offset: number;
+	length: number;
+
+	constructor(offset: number, length: number) {
+		this.offset = offset;
+		this.length = length;
+	}
+}
+
+interface DataHandler {
+	read(buffer: Uint8Array, offset: number): Promise<number>;
+}
+
+export class BPTreeNode {
+	public dataHandler: DataHandler;
+	public pointers: MemoryPointer[];
+	public keys: ReferencedValue[];
+
+	constructor(
+		dataHandler: DataHandler,
+		pointers: MemoryPointer[],
+		keys: ReferencedValue[]
+	) {
+		this.dataHandler = dataHandler;
+		this.pointers = pointers;
+		this.keys = keys;
+	}
+
+	addPointer(pointer: MemoryPointer) {
+		this.pointers.push(pointer);
+	}
+
+	addKey(key: ReferencedValue) {
+		this.keys.push(key);
+	}
```

Comment:
I don't think these are necessary because they're related to writing?

Reply:
Yeah - my bad.